### PR TITLE
More flexible Creative Tab API for MetaItems

### PR DESCRIPTION
**Pull request by Tictim**
## What
This PR refactors small portion of `MetaItem` class to add support for custom creative tabs on its variants. Notable features include:
 * Being able to change tabs of MetaItem itself - instead of copy-pasting the whole `getSubItems` declaration (like done in `MetaPrefixItem` previously), a method is available to change the base tab.
 * Being able to adjust the tab on `MetaValueItem` basis - it is possible to override the creative tabs for each MetaValueItem.

## Implementation Details
MetaValueItems with no creative tab overrides are added to tabs stored in field `MetaItem#defaultCreativeTabs`. The default tab itself can be set with `MetaItem#setCreativeTab()`/`MetaItem#setCreativeTabs()` methods.

`MetaValueItem#setCreativeTabs()` sets tab override to the MetaValueItem. Using empty array as an override makes the MetaValueItem hidden from all creative tabs; the internal implementation for `MetaValueItem#setInvisible()` and `MetaValueItem#isVisible()` has been changed to use overrides.

`MetaItem#additionalCreativeTabs` field is a union set of all additional tabs registered by `MetaValueItem#setCreativeTabs()`; the cache is made to quickly filter away irrelevant tabs in `getSubItems`. Additional creative tabs can be added to `MetaItem#addAdditionalCreativeTabs` manually, if needed.

`MetaValueItem#setInvisible(boolean)` method has been marked deprecated, as the method produced the opposite of what users might expect (calling `setInvisible(true)` made the item visible, and `setInvisible(false)` hid it). Instead of reversing the behavior, I've deprecated the method and created new method to preserve backwards compatibility. I think `setInvisibleIf` fits better on current implementation anyway, since it's not just setting `visible` bool field anymore; the new method name reflects the internal logic of hiding the item on input value `true`.

## Potential Compatibility Issues
There should be no compatibility issue on the implementation. Addons might need to adjust their MetaItem implementation to use the new API, if they overwrote `getSubItems` like in `MetaPrefixItem` previously.

### CODE REVIEW

File: src/main/java/gregtech/api/items/metaitem/MetaItem.java [hunk 1]
```diff
@@ -639,20 +642,45 @@ public ItemStack getContainerItem(@Nonnull ItemStack itemStack) {
...
-            if (!enabledItem.isVisible())
-                continue;
-            ItemStack itemStack = enabledItem.getStackForm();
-            enabledItem.getSubItemHandler().getSubItems(itemStack, tab, subItems);
+    }
+
+    @Override
+    protected boolean isInCreativeTab(CreativeTabs tab) {
+        return tab==CreativeTabs.SEARCH||
+                ArrayUtils.contains(defaultCreativeTabs, tab)||
```
Comment (TechLord22) [inline src/main/java/gregtech/api/items/metaitem/MetaItem.java:674]:
```suggestion
        return tab == CreativeTabs.SEARCH || 
                ArrayUtils.contains(defaultCreativeTabs, tab) || 
```

File: src/main/java/gregtech/common/items/MetaItem1.java [hunk 2]
```diff
@@ -224,26 +224,26 @@ public void registerSubItems() {
...
-        })).setInvisible(GTValues.HT);
-        ELECTRIC_PUMP_UXV = addItem(153, "electric.pump.uxv").addComponents(new TooltipBehavior(lines -> {
-            lines.add(I18n.format("metaitem.electric.pump.tooltip"));
-            lines.add(I18n.format("gregtech.universal.tooltip.fluid_transfer_rate", 1280 * 64 * 64 * 4 / 20));
-        })).setInvisible(GTValues.HT);
-        ELECTRIC_PUMP_OpV = addItem(154, "electric.pump.opv").addComponents(new TooltipBehavior(lines -> {
-            lines.add(I18n.format("metaitem.electric.pump.tooltip"));
-            lines.add(I18n.format("gregtech.universal.tooltip.fluid_transfer_rate", 1280 * 64 * 64 * 4 / 20));
-        })).setInvisible(GTValues.HT);
+        ELECTRIC_PUMP_UHV = addItem(150, "electric.pump.uhv").addComponents(new TooltipBehavior(lines15 -> {
```
Comment (TechLord22) [inline src/main/java/gregtech/common/items/MetaItem1.java:227]: For pumps, conveyors, and robot arms, these lambda parameters seemed to all have been renamed from `lines` to `lines#`. Is there a specific reason for this?

File: src/main/java/gregtech/api/items/metaitem/MetaItem.java [hunk 3]
```diff
@@ -639,20 +642,45 @@ public ItemStack getContainerItem(@Nonnull ItemStack itemStack) {
     @Nonnull
     @Override
     public CreativeTabs[] getCreativeTabs() {
-        return new CreativeTabs[]{GregTechAPI.TAB_GREGTECH, GregTechAPI.TAB_GREGTECH_MATERIALS};
+        if (additionalCreativeTabs.isEmpty()) return defaultCreativeTabs; // short circuit
+        Set<CreativeTabs> tabs = new ObjectArraySet<>();
+        tabs.addAll(Arrays.asList(defaultCreativeTabs));
+        tabs.addAll(additionalCreativeTabs);
```
Comment (TechLord22) [inline src/main/java/gregtech/api/items/metaitem/MetaItem.java:648]:
```suggestion
        Set<CreativeTabs> tabs = new ObjectArraySet<>(additionalCreativeTabs);
        tabs.addAll(Arrays.asList(defaultCreativeTabs));
```

File: src/main/java/gregtech/common/items/MetaItem1.java (hunk 2 above)
Comment (Tictim) [reply src/main/java/gregtech/common/items/MetaItem1.java:227]: I blame IntelliJ.

**Review by LAGIdiot — APPROVED** [review]
This change makes sense as it gives more options to addons. Implementation seems reasonable, I am not fan of `setInvisibleIf` but I understand reasoning behind it.

**Pull request merged**